WordPress/Classes/ViewRelated/Jetpack Branding/WPTabBarController+JetpackBanner.swift

```diff
@@ -0,0 +1,111 @@
+import UIKit
+
+extension WPTabBarController {
+
+    /// Adds a Jetpack powered banner on top of the main tab bar.
+    @objc func addJetpackBanner() {
+
+        guard AppConfiguration.isWordPress else {
+            return
+        }
+
+        jetpackBannerView = UIView()
+        jetpackBannerView.translatesAutoresizingMaskIntoConstraints = false
+
+
+        jetpackBannerView.backgroundColor = Appearance.jetpackBackgroundColor
+
+        view.insertSubview(jetpackBannerView, belowSubview: tabBar)
+
+        let jetpackButton = makeJetpackButton()
+        jetpackBannerView.addSubview(jetpackButton)
+
+        NSLayoutConstraint.activate([
+            jetpackBannerView.heightAnchor.constraint(equalToConstant: Appearance.jetpackBannerHeight),
+            jetpackBannerView.bottomAnchor.constraint(equalTo: tabBar.topAnchor),
+            jetpackBannerView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            jetpackBannerView.trailingAnchor.constraint(equalTo: view.trailingAnchor)
+        ])
+        jetpackBannerView.pinSubviewToAllEdges(jetpackButton)
+        NotificationCenter.default.post(name: .jetpackBannerToggled, object: true)
+
+    }
+
+    private func makeJetpackButton() -> UIButton {
+        let jetpackButton = UIButton()
+        jetpackButton.translatesAutoresizingMaskIntoConstraints = false
+        jetpackButton.setTitle(Appearance.jetpackBannerTitle, for: .normal)
+        jetpackButton.tintColor = .muriel(color: .jetpackGreen, .shade40)
+        jetpackButton.setTitleColor(UIColor(light: .black, dark: .white), for: .normal)
+        jetpackButton.titleLabel?.font = Appearance.jetpackButtonFont
+        jetpackButton.titleLabel?.adjustsFontForContentSizeCategory = true
+        jetpackButton.titleLabel?.minimumScaleFactor = Appearance.jetpackFontMinimumScaleFactor
+        jetpackButton.titleLabel?.adjustsFontSizeToFitWidth = true
+        jetpackButton.setImage(.gridicon(.plans, size: Appearance.jetpackIconSize), for: .normal)
+        jetpackButton.imageEdgeInsets = Appearance.jetpackIconInsets
+
+        // sets the background of the jp logo to white
+        if let imageView = jetpackButton.imageView {
+            let view = UIView()
+            view.backgroundColor = .white
+            view.translatesAutoresizingMaskIntoConstraints = false
+            jetpackButton.insertSubview(view, belowSubview: imageView)
+            view.layer.cornerRadius = Appearance.jetpackIconBackgroundSize / 2
+            view.clipsToBounds = true
+            NSLayoutConstraint.activate([
+                view.centerXAnchor.constraint(equalTo: imageView.centerXAnchor),
+                view.centerYAnchor.constraint(equalTo: imageView.centerYAnchor),
+                view.heightAnchor.constraint(equalToConstant: Appearance.jetpackIconBackgroundSize),
+                view.widthAnchor.constraint(equalToConstant: Appearance.jetpackIconBackgroundSize)
+            ])
+        }
+
+        return jetpackButton
```

Comment:
Is it intentional that you use an `UIButton`… but don't associate a target/action to it?

I didn't test on device in practice (only reading from the code), but wouldn't that make this tappable visually (as in, if you tap on the banner you're gonna have a visual feedback of the button being in a "pressed" state) while nothing would happen when tapped?